### PR TITLE
Remove Python 2 support

### PR DESCRIPTION
**Pull request by clalancette**
As discussed over in https://github.com/osrf/osrf_pycommon/pull/66#discussion_r444910962 , we should remove support for Python 2 from this package.  This PR does this by doing 3 different things:

1.  Removing trollius support
1.  Using modern asyncio keywords everywhere
1.  Changing the packaging so it only supports Python 3.5 and later

There is more work that can be done here; the indirection that was in place to support both Python 2 and Python 3 is probably no longer required.  But I think this is a good first step, and we can always clean some of that stuff up later.

### CODE REVIEW

File: tests/test_code_format.py [hunk 1]
```diff
@@ -18,9 +18,6 @@ def test_flake8():
     cmd.extend(['--ignore=C,D,Q,I'])
     # work around for https://gitlab.com/pycqa/flake8/issues/179
     cmd.extend(['--jobs', '1'])
-    if sys.version_info < (3, 4):
-        # Unless Python3, skip files with new syntax, like `yield from`
-        cmd.append('--exclude={0}/*async_execute_process_asyncio/impl.py'.format(source_dir))
```
Comment (clalancette) [inline tests/test_code_format.py:23]: @wjwwood said:

> These files would likely no longer be needed. And some of the files could be collapsed once there's only one implementation.

But I think you were thinking of this backwards.  We do need the `asyncio` implementation here, but in a Python 2 world we had to exclude those files from being flake8 checked.  Now that we are removing Python 2 support, we no longer need that exclusion.

But please correct me if I misunderstood you.

**Comment by clalancette** [comment]
@wjwwood Friendly ping here for review.

**Review by hidmic — APPROVED** [review]
LGTM with green CI

**Comment by wjwwood** [comment]
This is on me, because I need to do a final release with python support and then do a release that is python3 only without breaking older platforms and the pip install for python2. I messed this up for catkin tools and I just need to sit for a day or so and do it properly with test releases and stuff.

**Comment by wjwwood** [comment]
Oooook, I think we've got our last Python2 release out and working in `0.2.1` (it was released into rolling but not released as `python-osrf-pycommon` for ROS 1 until now). So now I can test this for merging (at least in ROS 2):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14931/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9673/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12606/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15095/)

**Comment by wjwwood** [comment]
CI looks good (Windows warning is not related, it's in `rclcpp_components`).

**Comment by clalancette** [comment]
:tada:

**Pull request merged**